state-chain/runtime/src/lib.rs

```diff
@@ -1423,6 +1423,23 @@ impl_runtime_apis! {
 			all_prewitnessed_swaps
 		}
 
+		fn cf_scheduled_swaps(base_asset: Asset, _quote_asset: Asset) -> Vec<(SwapLegInfo, BlockNumber)> {
+
+			let first_block = Swapping::first_unprocessed_block();
+			let last_block = System::block_number() + pallet_cf_swapping::SWAP_DELAY_BLOCKS;
+
+			debug_assert!(first_block < last_block);
+
+			(first_block..=last_block).flat_map(|block| {
```

Comment:
You could just read the entire SwapQueue map into memory (Instead of one element at a time), then you can make this code much simpler.